std/typecons.d

```diff
@@ -6414,7 +6415,7 @@ if (!is(T == class) && !(is(T == interface)))
             import core.lifetime : emplace, forward;
 
             allocateStore();
-            version (D_Exceptions) scope(failure) deallocateStore();
+            version (D_Exceptions) scope(failure) () @trusted { deallocateStore(); }();
             emplace(&_store._payload, forward!args);
```

Comment:
The deallocation can't be trusted if the `emplace` call isn't pure.

Reply:
I don't understand the correlation, can you elaborate?

Reply:
Impure `emplace` (i.e. an impure constructor call within it) may store a pointer somewhere, making use after free possible.